src/models/TableModel/impl/InMemoryTable.ts

```diff
@@ -27,37 +43,80 @@ export const createTablesFromCx = (id: IdType, cx: Cx2): [Table, Table] => {
     Record<string, CxValue>
   > = cxUtil.getEdgeAttributes(cx)
 
-  nodeAttr.forEach((attr, nodeId) => {
-    nodeTable.rows.set(nodeId, attr as Record<AttributeName, ValueType>)
-  })
-
-  edgeAttr.forEach((attr, edgeId) => {
-    edgeTable.rows.set(edgeId, attr as Record<AttributeName, ValueType>)
-  })
+  // append position specific data to columns/rows
+  const positionXKey = 'positionX' // todo this can conflict with exiting property keys
+  // generate a unique key for position and also need some thought into where positions should go
+  const positionYKey = 'positionY'
+  const cxIdKey = 'cxId'
 
   // Columns
   const attrDec = cxUtil.getAttributeDeclarations(cx)
   const attrDefs: AttributeDeclaration = attrDec.attributeDeclarations[0]
-  const { nodes, edges } = attrDefs
 
-  const nodeAttrNames: string[] = Object.keys(nodes)
-  const edgeAttrNames = Object.keys(edges)
+  Object.entries(attrDefs.nodes).forEach(([attrName, attrDef]) => {
+    const columnDef: Column = {
+      type: attrDef.d as ValueTypeName,
+      name: attrName,
+      defaultValue: (attrDef.v ?? valueTypeDefaults[attrDef.d]) as ValueType,
+      ...(attrDef.a != null ? { alias: attrDef.a } : {}),
+    }
+
+    nodeTable.columns.set(attrName, columnDef)
+  })
+  nodeTable.columns.set(positionXKey, {
+    type: 'double',
+    name: positionXKey,
+    defaultValue: 0.0,
+  })
+  nodeTable.columns.set(positionYKey, {
+    type: 'double',
+    name: positionYKey,
+    defaultValue: 0.0,
+  })
+  nodeTable.columns.set(cxIdKey, {
+    type: 'string',
+    name: cxIdKey,
+    defaultValue: '',
+  })
+
+  Object.entries(attrDefs.edges).forEach(([attrName, attrDef]) => {
+    const columnDef = {
+      type: attrDef.d as ValueTypeName,
+      name: attrName,
+      defaultValue: (attrDef.v ?? valueTypeDefaults[attrDef.d]) as ValueType,
+      ...(attrDef.a != null ? { alias: attrDef.a } : {}),
+    }
+
+    edgeTable.columns.set(attrName, columnDef)
+  })
+  edgeTable.columns.set(cxIdKey, {
+    type: 'string',
+    name: cxIdKey,
+    defaultValue: '',
+  })
+
+  // Rows
+  const cxNodes: Node[] = cxUtil.getNodes(cx)
 
-  nodeAttrNames.forEach((attrName) => {
-    nodeTable.columns.set(attrName, nodes[attrName].d as ValueTypeName)
+  nodeAttr.forEach((attr, nodeId) => {
+    const posX = (cxNodes[+nodeId]?.x ?? 0) as ValueType
+    const posY = (cxNodes[+nodeId]?.y ?? 0) as ValueType
+    attr[positionXKey] = posX
+    attr[positionYKey] = posY
+    attr[cxIdKey] = nodeId
+    nodeTable.rows.set(nodeId, attr as Record<AttributeName, ValueType>)
   })
-  edgeAttrNames.forEach((attrName) => {
-    edgeTable.columns.set(attrName, edges[attrName].d as ValueTypeName)
+
+  edgeAttr.forEach((attr, edgeId) => {
```

Comment:
Translate edgeId to cyjs compatible edge id (10 -> e10), also make a function that is shared between createNetworkFromCX and createTableFromCX and createVisualStyleFromCX